components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/certificatechecktrigger/certificatechecktrigger_test.go

```diff
@@ -0,0 +1,301 @@
+	ipConfig := models.Config{}
+
+	json.Unmarshal([]byte(`{
+		  "ssh_user": {
+			"user_name": "ubuntu",
+			"private_key": "test_key",
+			"sudo_password": "test@123"
+		  },
+		  "arch": "existing_nodes",
+		  "backup": {
+			"file_system": {
+			  "mount_location": "/mnt/automate_backups"
+			}
+		  },
+		  "hardware": {
+			"automate_node_count": 1,
+			"automate_node_ips": [
+			  "1.2.3.4"
+			],
+			"chef_infra_server_node_count": 1,
+			"chef_infra_server_node_ips": [
+			  "5.6.7.8"
+			],
+			"postgresql_node_count": 1,
+			"postgresql_node_ips": [
+			  "9.10.11.12"
+			],
+			"opensearch_node_count": 1,
+			"opensearch_node_ips": [
+			  "14.15.16.17"
+			]
+		  },
+		  "certificate": {
+			"fqdn":"my_fqdn",
+			"root_cert":"---- VALID ROOT CA ----",
+			"nodes":[
+				{
+					"ip":"1.2.3.4",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"5.6.7.8",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"9.10.11.12",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"14.15.16.17",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				}
+			]
+		  }
+		}`), &ipConfig)
+	return ipConfig
+}
+
+func GetCertificateRequest() models.CertificateCheckRequest {
+	req := models.CertificateCheckRequest{}
+	json.Unmarshal([]byte(`{
+		"RootCertificate": "---- VALID ROOT CA ----",
+		"PrivateKey": "---- VALID PRIVATE KEY ----",
+		"NodeCertificate": "---- VALID NODE CERT ----",
+		"AdminPrivateKey": "---- VALID ADMIN PRIVATE KEY ----",
+		"AdminCertificate": "---- VALID ADMIN CERT ----"
+	 }`), &req)
+	return req
+}
+
+func TestNewCertificateCheck(t *testing.T) {
+	testPort := "1234"
+	cc := NewCertificateCheck(logger.NewTestLogger(), testPort)
+	assert.NotNil(t, cc)
+	assert.NotNil(t, cc.log)
+	assert.Equal(t, constants.LOCAL_HOST_URL, cc.host)
+	assert.Equal(t, testPort, cc.port)
+}
+
+func startMockServerOnCustomPort(mockServer *httptest.Server, port string) error {
+	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%s", port))
+	if err != nil {
+		return err
+	}
+	mockServer.Listener = l
+	mockServer.Start()
+	return nil
+}
+
+func TestCertificateCheck_Run(t *testing.T) {
+
+	t.Run("Returns OK", func(t *testing.T) {
+		//starting the mock server on custom port
+		mockServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
```

Comment:
since this is a post request, please assert the input request as well in the mock server.

Reply:
Added as discussed on the call